minetest/minetest
# InventoryManager: Disallow resizing or deleting inventory lists that are in use

Naive solution to prevent InventoryList UAF and OOB ItemStack access

Fixes #13358 by preventing shrink/clear operations on InventoryLists that are in use by the inventory action.

## To do

This PR is Ready for Review.

## How to test

1. https://github.com/cornernote/minetest-workbench/
2. Place a workbench
3. Open
4. Move items
5. Minetest no longer segfaults occasionally but throws a proper error,

I do not know when Minetest became incompatible with this mod, but I assume it was caused by the callback fixes years ago.
